src/firebase.js

```diff
@@ -13,79 +13,94 @@ const firebaseConfig = {
+          .update({
+            totalSessionCount: log.totalSessionCount + 1,
+            totalTime: log.totalTime + parseInt(sessionLength)
+          });
+      }
+    });
 };
 
-export const saveUserSettings = (session, shortBreak, longBreak) => {
-  firebase.auth().onAuthStateChanged(user => {
-    if (user) {
-      // User is signed in.
-      db.collection('settings')
-        .where('userId', '==', user.uid)
-        .get()
-        .then(response => {
-          const { empty, docs } = response;
-          if (empty) {
-            db.collection('settings').add({
-              userId: user.uid,
-              sessionLength: session,
-              shortBreakLength: shortBreak,
-              longBreakLength: longBreak
-            });
-          } else {
-            const [doc] = docs;
-            const userSettings = {
-              id: doc.id,
-              sessionLength: doc.data().sessionLength,
-              shortBreakLength: doc.data().shortBreakLength,
-              longBreakLength: doc.data().longBreakLength
-            };
-            db.collection('settings')
-              .doc(userSettings.id)
-              .update({
-                sessionLength: session,
-                shortBreakLength: shortBreak,
-                longBreakLength: longBreak
-              });
-          }
+export const saveUserSettings = (
+  userId,
+  { sessionLength, shortBreakLength, longBreakLength }
+) => {
+  db.collection('settings')
+    .where('userId', '==', userId)
+    .get()
+    .then(response => {
+      const { empty, docs } = response;
+      if (empty) {
+        db.collection('settings').add({
+          userId: userId,
+          sessionLength,
+          shortBreakLength,
+          longBreakLength
         });
-    }
-  });
+      } else {
+        const [doc] = docs;
+        const userSettings = {
+          id: doc.id,
+          sessionLength: doc.data().sessionLength,
+          shortBreakLength: doc.data().shortBreakLength,
+          longBreakLength: doc.data().longBreakLength
+        };
+        db.collection('settings')
+          .doc(userSettings.id)
+          .update({
+            sessionLength,
+            shortBreakLength,
+            longBreakLength
+          });
+      }
+    });
+};
+
+export const loadUserSettings = userId => {
+  db.collection('settings')
+    .where('userId', '==', userId)
+    .get()
+    .then(snapshot => {
+      const settings = snapshot.docs.map(doc => ({
+        id: doc.id,
+        userId: doc.data().userId,
+        sessionLength: doc.data().sessionLength,
+        shortBreakLength: doc.data().shortBreakLength,
+        longBreakLength: doc.data().longBreakLength
+      }));
+      const [setting] = settings;
+      this.setState({
+        seconds: Number(setting.sessionLength),
+        sessionLength: Number(setting.sessionLength),
+        shortBreakLength: Number(setting.shortBreakLength),
+        longBreakLength: Number(setting.longBreakLength)
+      });
```

Comment:
This is a big no-no.  You can't really know to what `this` refers to. This function should only return the data, or in this case a promise with the data